src/sim-host/protocol/socket.js

```diff
@@ -11,52 +11,68 @@ module.exports.initialize = function (pluginHandlers, serviceToPluginMap) {
     socket.on('init-telemetry', function (data) {
         telemetry.init(socket);
     });
-    socket.on('exec', function (data) {
-        if (!data) {
-            throw 'Exec called on simulation host without exec info';
-        }
-
-        var index = data.index;
-        if (typeof index !== 'number') {
-            throw 'Exec called on simulation host without an index specified';
-        }
-
-        var success = data.hasSuccess ? getSuccess(index) : null;
-        var failure = data.hasFail ? getFailure(index) : null;
-
-        var service = data.service;
-        if (!service) {
-            throw 'Exec called on simulation host without a service specified';
-        }
-
-        var action = data.action;
-        if (!action) {
-            throw 'Exec called on simulation host without an action specified';
-        }
-
-        console.log('Exec ' + service + '.' + action + ' (index: ' + index + ')');
-
-        var handler = pluginHandlers[service] && pluginHandlers[service][action];
-        var telemetryProps = { plugin: serviceToPluginMap && serviceToPluginMap[service], service: service, action: action };
-        if (!handler) {
-            telemetryProps.handled = 'none';
-            handler = pluginHandlers['*']['*'];
-            handler(success, failure, service, action, data.args);
-        } else {
-            telemetryProps.handled = 'sim-host';
-            handler(success, failure, data.args);
-        }
-
-        telemetry.sendClientTelemetry('exec', telemetryProps);
-    });
+
     socket.on('refresh', function () {
         document.location.reload(true);
     });
+
+    socket.on('app-plugin-list', function (data) {
+        // TODO: process the list of plugins
```

Comment:
Please open an issue so we don't forget about this `TODO` if it is still needed.